pkg/autoscaler/autoscaler.go

```diff
@@ -204,25 +208,48 @@ func remaining(t time.Time) time.Duration {
 }
 
 var _ = Describe("[Feature:Machines] Autoscaler should", func() {
+
+	var workloadMemRequest resource.Quantity
+	var client runtimeclient.Client
+	var err error
+
+	BeforeEach(func() {
+		client, err = framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		workerNodes, err := framework.GetWorkerNodes(client)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(workerNodes)).To(BeNumerically(">=", 1))
+
+		memCapacity := workerNodes[0].Status.Capacity[corev1.ResourceMemory]
+		Expect(memCapacity).ShouldNot(BeNil())
+		Expect(memCapacity.String()).ShouldNot(BeEmpty())
+		glog.Infof("Memory capacity of worker node %q is %s", workerNodes[0].Name, memCapacity.String())
+
+		bytes, ok := memCapacity.AsInt64()
+		Expect(ok).Should(BeTrue())
+
+		// 70% - enough that the existing and new nodes will
+		// be used, not enough to have more than 1 pod per
+		// node.
+		workloadMemRequest = resource.MustParse(fmt.Sprintf("%v", 0.7*float32(bytes)))
+	})
+
+	ctx := context.Background()
 	cascadeDelete := metav1.DeletePropagationForeground
+	deleteObject := func(name string, obj runtime.Object) error {
```

Comment:
can't we add and `afterEach` clause which runs the logic in https://github.com/openshift/cluster-api-actuator-pkg/pull/148/files#diff-a6c4792dc41958d712d5dc9615145d28R484-R488 so we clean up consistently every validation?

Reply:
yeah, i think we definitely can. i'll add it

Reply:
i'm looking into this deeper and i think it will require some restructuring to have a common cleanup routine.